src/mon/OSDMonitor.cc

```diff
@@ -496,7 +496,8 @@ void OSDMonitor::handle_conf_change(const ConfigProxy& conf,
 {
   dout(10) << __func__ << " " << changed << dendl;
 
-  if (changed.count("mon_memory_autotune")) {
+  if (changed.count("mon_memory_autotune") ||
+      g_conf()->mon_memory_autotune) {
     _set_cache_autotuning();
   }
   if (changed.count("mon_memory_target") ||
```

Comment:
How about the following alternative that achieves the same result? I think this helps silo the purpose of the  `if` blocks in `handle_conf_change()`. What do you think?

```suggestion
  if (changed.count("mon_memory_target") ||$
    changed.count("rocksdb_cache_size")) {
  _set_cache_autotuning();
  int r = _update_mon_cache_settings();$  
  if (r < 0) {$                           
```

Reply:
> How about the following alternative that achieves the same result? I think this helps silo the purpose of the `if` blocks in `handle_conf_change()`. What do you think?

Well, that would mean _set_cache_autotuning() is guaranteed to be called right? If that's the case why not just remove the if statement around mon_memory_autotune and let it be called any time we enter the function?

Reply:
I wouldn't prefer the if condition to be removed as `_set_cache_autotuning`  may end up being called unnecessarily for unrelated/unknown config changes. I am okay with your original change or my suggested change. I am therefore approving.

Reply:
> I wouldn't prefer the if condition to be removed as `_set_cache_autotuning` may end up being called unnecessarily for unrelated/unknown config changes. I am okay with your original change or my suggested change. I am therefore approving.

Yes, you are right of course, I only considered this later :) I'm happy to go with your change after I test it.